src/decorator.ts

```diff
@@ -38,25 +36,62 @@ export class Decorator {
       lineProp.overviewRulerColor = ruler.trim();
     }
 
+
+    this.lineDecorator = vscode.window.createTextEditorDecorationType(lineProp);
+    this.maybeReloadGutterDecorator(Severity.Default);
+  }
+
+  maybeReloadGutterDecorator(newSeverity: Severity) {
+    const gutterProp: vscode.DecorationRenderOptions = {};
+    const config: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration();
+
     const showGutterIcon: boolean | undefined = config.get(
       "linenote.showGutterIcon"
     );
-    if (showGutterIcon) {
-      let iconPath: string | undefined = config.get("linenote.gutterIconPath");
-      if (iconPath) {
-        gutterProp.gutterIconPath = iconPath;
-      } else {
-        gutterProp.gutterIconPath = this.context.asAbsolutePath(
-          "images/gutter.png"
-        );
+
+    if (this.severity === undefined || this.severity !== newSeverity) {
+      // since we're going to switch decorators, dispose if needed
+      if (this.gutterDecorator) {
+        this.gutterDecorator.dispose();
       }
-      gutterProp.gutterIconSize = "cover";
-    }
 
-    this.lineDecorator = vscode.window.createTextEditorDecorationType(lineProp);
-    this.gutterDecorator = vscode.window.createTextEditorDecorationType(
-      gutterProp
-    );
+      switch (newSeverity) {
+        case Severity.Low:
+          gutterProp.gutterIconSize = "cover";
+          gutterProp.gutterIconPath = this.context.asAbsolutePath(
+            "images/gutter-low.png"
+          );
+          break;
+        case Severity.Medium:
+          gutterProp.gutterIconSize = "cover";
+          gutterProp.gutterIconPath = this.context.asAbsolutePath(
+            "images/gutter-medium.png"
+          );
+          break;
+        case Severity.High:
+          gutterProp.gutterIconSize = "cover";
+          gutterProp.gutterIconPath = this.context.asAbsolutePath(
+            "images/gutter-high.png"
+          );
+          break;
+        default:
+          if (showGutterIcon) {
```

Comment:
I'm not sure if this is the right behavior. I can see two possibilities:
- the `showGutterIcon` setting should be absolute & final, i.e. NO gutter icons whatsoever
- OR the `showGutterIcon` should be for default icons. If the user adds severity hashtags in the note, the presence of the tag overrides the `showGutterIcon` setting.

I've implemented the latter.